tooltester/src/main/java/io/dockstore/tooltester/client/cli/Client.java

```diff
@@ -322,12 +337,56 @@ private void setUpWorkflowApi() {
         defaultApiClient.setBasePath(this.workflowRunnerConfig.getServerUrl());
         setWorkflowsApi(new WorkflowsApi(defaultApiClient));
     }
-    private void runToolTesterOnWorkflows(String pathToConfigFile) throws InterruptedException {
+
+    private List<File> getAllFilesInDirectory(File directory) {
+        List<File> filesInDirectory = new ArrayList<>();
+        for (File file: directory.listFiles()) {
+            if (file.isDirectory()) {
+                filesInDirectory.addAll(getAllFilesInDirectory(file));
+            } else {
+                filesInDirectory.add(file);
+            }
+        }
+        return filesInDirectory;
+    }
+
+    private void uploadResults(String urlToUploadTo, String resultsDirectoryString, String pathToConfigFile) {
+        this.workflowRunnerConfig = new WorkflowRunnerConfig(pathToConfigFile);
+        setUpExtendedGa4GhApi(urlToUploadTo);
+        Gson gson = new Gson();
+
+        Path resultsDirectoryPath = Path.of(resultsDirectoryString);
+        File resultsDirectory = new File(resultsDirectoryString);
+        List <File> allResultFiles = getAllFilesInDirectory(resultsDirectory);
+        for (File file: allResultFiles) {
+
+            Path path = Path.of(file.getPath());
+            String fileContent = null;
+            try {
+                fileContent = Files.readString(path);
+            } catch (IOException e) {
+                exceptionMessage(e, "There was an error reading " + path + " to a string", IO_ERROR);
+            }
+
+            final RunExecution runMetrics = gson.fromJson(fileContent, RunExecution.class);
+
+            final String fileKey = resultsDirectoryPath.relativize(path).toString();
+
+            final String toolID = getToolId(fileKey);
+            final String versionName = getVersionName(fileKey);
+            final String metricsPlatform = getMetricsPlatform(fileKey);
+
+            LOG.info("Uploading run metrics for: " + fileKey);
```

Comment:
Could also log the environment it's being uploaded to

Reply:
I don't think this is needed, as if it is possible for the metrics to be uploaded, it means the workflows exists in both environments (environment tooltester ran in and environment uploaded to).